src/Optimization/UI/AxisymmetricOptimization.jl

```diff
@@ -0,0 +1,349 @@
+# MIT license
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# See LICENSE in the project root for full license information.
+
+module AxisymmetricOptimization
+
+using OpticSim, OpticSim.Geometry, OpticSim.Emitters
+using CSV, DataFrames
+using Ipopt
+using JuMP
+import DataStructures
+
+global global_model = nothing
+global global_ax = nothing
+global global_df = nothing
+global global_counter = 0
+global solution_history = []
+global statistics = Dict()
+global options = Dict()
+
+function set_global_ax(ax) 
+    global global_ax = ax
+end 
+
+global emitter = Sources.Source(origins=Emitters.Origins.Hexapolar(5, 8.0, 8.0), directions=Directions.Constant(), transform=Transform(Vec3(0.0,0.0,5.0), Geometry.unitZ3()*-1))
+
+mutable struct VarInfo
+    _row::Int
+    _col::Int
+    _name::String
+    _start_value::Float64
+    _lower_bound::Union{Float64,Nothing}
+    _upper_bound::Union{Float64,Nothing}
+    _info::Dict
+end
+
+function get_number(dic, key)
+    v = get(dic, key, Nothing)
+    if (v != Nothing)
+        v = parse(Float64, v)
+    end
+    return v
+end
+
+function try_parse_variable_1(row, col, str)
+
+    float_re = "[+-]?(?:[0-9]+(?:[.][0-9]*)?|[.][0-9]+)"
+
+    re = Regex(
+        "^\\s*"*                                    # white spaces at start
+        "(?:"*
+            "(?<lower>(?:$float_re))" *             # lower bound
+            "\\s*"*                                 # white spaces 
+            "(?<lower_cond><=?)" *                  # lower condition
+        ")?"*
+        "\\s*"*                                     # white spaces 
+        "(?<name>(?:[@a-zA-Z][a-zA-Z0-9_@]*))" *    # variable name
+        "\\s*"*                                     # white spaces 
+        "(?<assignment>=)" *                        # assignment
+        "\\s*"*                                     # white spaces 
+        "(?<middle>(?:$float_re))" *                # starting value
+        "\\s*"*                                     # white spaces 
+        "(?:"*
+            "(?<upper_cond><=?)" *                  # uppwe condition
+            "\\s*"*                                 # white spaces 
+            "(?<upper>(?:$float_re))" *             # upper bound
+        ")?"*
+        "\\s*\$"                                    # white spaces at end
+    )
+    
+    m=match(re,str)
+    if (m === nothing)
+        return nothing
+    end
+
+    name = m["name"]
+    start = parse(Float64, m["middle"])
+    lower = m["lower"]===nothing ? nothing : parse(Float64, m["lower"])
+    upper = m["upper"]===nothing ? nothing : parse(Float64, m["upper"])
+
+    var_info = VarInfo(row, col, name, start, lower, upper, Dict())
+
+    return var_info;
+end
+
+
+struct Axisymmetric
```

Comment:
This isn't very descriptive. Would AxisymmetricOptimization be more appropriate?